### PR TITLE
Added support for new -SkipDuplicate option in NuGet 5.1

### PR DESCRIPTION
**Pull request by jacksga**
Related to: [Push Command option to Skip Existing Packages](https://github.com/NuGet/NuGet.Client/pull/2802)

### CODE REVIEW

File: src/NuGet.Server.Core/Core/DuplicatePackageException.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace NuGet.Server.Core
+{
+    /// <summary>
+    /// This exception is thrown when trying to add a package in a version that already exists on the server 
+    /// and <see cref="NuGet.Server.Core.Infrastructure.ServerPackageRepository.AllowOverrideExistingPackageOnPush"/> is set to false.
+    /// </summary>
+    public class DuplicatePackageException : InvalidOperationException
```
Comment (loic-sharma) [inline src/NuGet.Server.Core/Core/DuplicatePackageException.cs]: Does this need to extend `InvalidOperationException`? I wouldn't extend `InvalidOperationException` unless necessary

File: src/NuGet.Server.V2/Controllers/NuGetODataController.cs [hunk 2]
```diff
@@ -403,8 +403,16 @@ public virtual async Task<HttpResponseMessage> UploadPackage(CancellationToken t
...
             if (_authenticationService.IsAuthenticated(User, apiKey, package.Id))
             {
-                await _serverRepository.AddPackageAsync(package, token);
-                retValue = Request.CreateResponse(HttpStatusCode.Created);
+                try
+                {
+                    await _serverRepository.AddPackageAsync(package, token);
+                    retValue = Request.CreateResponse(HttpStatusCode.Created);
+
+                }
```
Comment (loic-sharma) [inline src/NuGet.Server.V2/Controllers/NuGetODataController.cs]: Please remove the empty line on line 410.

**Comment by loic-sharma** [comment]
Looks great, thanks for doing this!

File: src/NuGet.Server.Core/Core/DuplicatePackageException.cs (hunk 1 above)
Comment (jacksga) [reply src/NuGet.Server.Core/Core/DuplicatePackageException.cs]: I did this thinking that it still is an invalid operation thus I wanted to "keep it in there". But I will remove it if you wish.

File: src/NuGet.Server.V2/Controllers/NuGetODataController.cs (hunk 2 above)
Comment (jacksga) [reply src/NuGet.Server.V2/Controllers/NuGetODataController.cs]: Done!

**Comment by joelverhagen** [comment]
Nice work. Could you add an integration test to [test/NuGet.Server.Tests/IntegrationTests.cs](https://github.com/NuGet/NuGet.Server/blob/master/test/NuGet.Server.Tests/IntegrationTests.cs)? This will cover the assertion that we return 409 and this case is generally a good thing to test.

**Comment by jacksga** [comment]
I added the test, is there anything else?

**Comment by joelverhagen** [comment]
> I added the test, is there anything else?

Nope, that's all. Great work @jacksga!

**Comment by joelverhagen** [comment]
This will be released as 3.4.0.

**Comment by joelverhagen** [comment]
This is now available on nuget.org under version 3.4.0. Thanks again!

**Pull request merged**